### PR TITLE
add test for decimal to decimal

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

- add test for decimal to decimal

- move decimal test together

part of #1443 

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/src/physical_plan/expressions/cast.rs [hunk 1]
```diff
@@ -269,6 +269,168 @@ mod tests {
...
+                None,
+            ],
+            DEFAULT_DATAFUSION_CAST_OPTIONS
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_cast_decimal_to_numeric() -> Result<()> {
```
Comment (liukun4515) [inline datafusion/src/physical_plan/expressions/cast.rs:326]: move test to here

File: datafusion/src/physical_plan/expressions/try_cast.rs [hunk 2]
```diff
@@ -227,6 +227,155 @@ mod tests {
...
+                Some(500_i128),
+                None,
+            ]
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_try_cast_decimal_to_numeric() -> Result<()> {
```
Comment (liukun4515) [inline datafusion/src/physical_plan/expressions/try_cast.rs:270]: move test to here

File: datafusion/src/physical_plan/expressions/try_cast.rs [hunk 3]
```diff
@@ -227,6 +227,155 @@ mod tests {
         }};
     }
 
+    #[test]
+    fn test_try_cast_decimal_to_decimal() -> Result<()> {
```
Comment (liukun4515) [inline datafusion/src/physical_plan/expressions/try_cast.rs:231]: add test for decimal to decimal

File: datafusion/src/physical_plan/expressions/cast.rs [hunk 4]
```diff
@@ -269,6 +269,168 @@ mod tests {
...
+        let mut decimal_builder = DecimalBuilder::new(array.len(), precision, scale);
+        for value in array {
+            decimal_builder.append_value(*value)?
+        }
+        decimal_builder.append_null()?;
+        Ok(decimal_builder.finish())
+    }
+
+    #[test]
+    fn test_cast_decimal_to_decimal() -> Result<()> {
```
Comment (liukun4515) [inline datafusion/src/physical_plan/expressions/cast.rs:286]: add test for decimal to decimal

**Review by alamb — APPROVED** [review]

**Pull request merged**